binary_search_tree/tree.py

```diff
@@ -14,44 +14,160 @@ class Tree:
+                current_node = current_node.right
+        return None
 
-    # Time Complexity: 
-    # Space Complexity: 
+  
+  
+    # Time Complexity: O(logn)
+    # Space Complexity: O(1)
     def inorder(self):
-        pass
+        traversal_list = []
+        if not self.root:
+            return traversal_list
+        self.inorder_helper(self.root, traversal_list)
+        return traversal_list
+    
+    # Left, Root, Right 
+    def inorder_helper(self, node, traversal_list):
+        if node == None:
+            return traversal_list
+        self.inorder_helper(node.left, traversal_list)
+        
+        traversal_list.append({
+            'key': node.key,
+            'value' : node.value
+        })
+        self.inorder_helper(node.right, traversal_list)
+        return traversal_list
+    
+   
 
-    # Time Complexity: 
-    # Space Complexity:     
+    
+    #Root, Left, Right
+    def preorder_helper(self, node, traversal_list):
+        
+        if node == None:
+            return traversal_list
+        traversal_list.append({'key': node.key,
+                            'value' : node.value})
+           
+        self.preorder_helper(node.left, traversal_list)
+        self.preorder_helper(node.right, traversal_list)
+        return traversal_list  
+    
+    # Time Complexity: O(logn)
+    # Space Complexity: O(1)    
     def preorder(self):
-        pass
-
-    # Time Complexity: 
-    # Space Complexity:     
+        traversal_list = []
+        if not self.root:
+            return traversal_list
+        self.preorder_helper(self.root, traversal_list)
+        return traversal_list
+    
+    
+   
+    # Left, Right, Root
+    def postorder_helper(self, node, traversal_list):
+        
+        if node == None:
+            return traversal_list
+            
+        self.postorder_helper(node.left, traversal_list)
+        
+        self.preorder_helper(node.right, traversal_list)
+        
+        traversal_list.append({'key': node.key,
+                            'value' : node.value})
+        return traversal_list
+    
+    # Time Complexity: O(logn)
+    # Space Complexity: O(1) 
     def postorder(self):
-        pass
+        traversal_list = []
+        if not self.root:
+            return traversal_list
+        self.postorder_helper(self.root, traversal_list)
+        return traversal_list
 
     # Time Complexity: 
     # Space Complexity:     
+    # to find height we need to do DFS and store a value after each leaf is found. 
+    def height_helper(self, node):
+        if node == None:
+            return 0
+        count = 1
+        left_height = self.height_helper(node.left)
+        right_height = self.height_helper(node.right)
+        return max(left_height,right_height)+ 1
+    
     def height(self):
-        pass
+        if not self.root:
+            return 0
+        return self.height_helper(self.root)
+        
 
+    def bfs_helper(self,node,bfs_list):
```

Comment:
⚠️ Good start, but for BFS you really need a queue of some sort, so recursion isn't the best solution.